sympy/polys/polyclasses.py

```diff
@@ -539,9 +539,11 @@ def exquo(f, g):
         """Computes polynomial exact quotient of ``f`` and ``g``. """
         lev, dom, per, F, G = f.unify(g)
         res = per(dmp_exquo(F, G, lev, dom))
-        if f.ring and res not in f.ring:
-            from sympy.polys.polyerrors import ExactQuotientFailed
-            raise ExactQuotientFailed(f, g, f.ring)
+        if f.ring:
+            if res not in f.ring:
+                raise RuntimeError("exquo...")
+                from sympy.polys.polyerrors import ExactQuotientFailed
+                raise ExactQuotientFailed(f, g, f.ring)
```

Comment:
I think that these are the only places where the `DMP.ring` attribute is used for anything nontrivial. I want to remove them all and then remove the ring attribute and then simplify `DMP.unify` and remove `per`.